module/programs/helper.js

```diff
@@ -150,6 +197,827 @@ module.exports = class ProgramsHelper {
+          };
+        }
+
+        return resolve({
+          message: messageConstants.apiResponses.ENTITIES_ADDED_IN_PROGRAM,
+          success: true,
+        });
+      } catch (error) {
+        return resolve({
+          success: false,
+          status: error.status
+            ? error.status
+            : httpStatusCode["internal_server_error"].status,
+          message: error.message,
+        });
+      }
+    });
+  }
+
+  /**
+   * remove roles in program.
+   * @method
+   * @name removeRolesInScope
+   * @param {String} programId - Program Id.
+   * @param {Array} roles - roles data.
+   * @returns {JSON} - Added roles data.
+   */
+
+  static removeRolesInScope(programId, roles) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        let programData = await programsQueries.programDocuments(
+          {
+            _id: programId,
+            scope: { $exists: true },
+            isAPrivateProgram: false,
+          },
+          ["_id"]
+        );
+
+        if (!(programData.length > 0)) {
+          return resolve({
+            status: httpStatusCode.bad_request.status,
+            message: messageConstants.apiResponses.PROGRAM_NOT_FOUND,
+          });
+        }
+
+        let userRoles = await userRolesHelper.roleDocuments(
+          {
+            code: { $in: roles },
+          },
+          ["_id", "code"]
+        );
+
+        if (!(userRoles.length > 0)) {
+          return resolve({
+            status: httpStatusCode.bad_request.status,
+            message: messageConstants.apiResponses.INVALID_ROLE_CODE,
+          });
+        }
+
+        let updateProgram = await programsQueries
+          .findOneAndUpdate(
+            {
+              _id: programId,
+            },
+            {
+              $pull: { "scope.roles": { $in: userRoles } },
+            },
+            { new: true }
+          )
+          ;
+
+        if (!updateProgram || !updateProgram._id) {
+          throw {
+            message: messageConstants.apiResponses.PROGRAM_NOT_UPDATED,
+          };
+        }
+
+        return resolve({
+          message: messageConstants.apiResponses.ROLES_REMOVED_IN_PROGRAM,
+          success: true,
+        });
+      } catch (error) {
+        return resolve({
+          success: false,
+          status: error.status
+            ? error.status
+            : httpStatusCode["internal_server_error"].status,
+          message: error.message,
+        });
+      }
+    });
+  }
+  
+  /**
+   * remove entities in program scope.
+   * @method
+   * @name removeEntitiesInScope
+   * @param {String} programId - Program Id.
+   * @param {Array} entities - entities.
+   * @returns {JSON} - Removed entities data.
+   */
+
+  static removeEntitiesInScope(programId, entities) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        let programData = await programsQueries.programDocuments(
```

Comment:
@praveenKDass please add comments

Reply:
Covering this in entity management PR review @VISHNUDAS-tunerlabs